backend/papersfeed/utils/recommendations/utils.py

```diff
@@ -54,6 +75,58 @@ def insert_user_recommendation(args):
         modification_date__lt=(datetime.now() + timedelta(hours=-2))
     ).delete()
 
+def select_recommendation(args):
+    """Get Recommendations"""
+
+    request_user = args[constants.USER]
+    page_number = 1 if constants.PAGE_NUMBER not in args else int(args[constants.PAGE_NUMBER])
+
+    recommendation_queryset = UserRecommendation.objects.filter(user_id=request_user.id)
+
+    recommendations = get_results_from_queryset(recommendation_queryset, 5, page_number)
+
+    is_finished = not recommendations.has_next()
+
+    recommendations = __pack_recommendations(recommendations, request_user)
+
+    return recommendations, page_number, is_finished
+
+def select_paper_all(args):
+    """Get All Papers"""
+
+    page_number = 1 if constants.PAGE_NUMBER not in args else int(args[constants.PAGE_NUMBER])
+
+    paper_queryset = Paper.objects.all().annotate(
+        ItemId=F('id')
+    ).values(
+        'ItemId'
+    )
+
+    papers = Paginator(paper_queryset, 1000).get_page(page_number)
```

Comment:
[Just for check] So, in ML codes, is there a part getting papers repetitively? And did you check if getting 1000 papers is stable?

Reply:
I should make the code for getting all papers and it would be used once. I checked just more than 800 papers are transmitted well.

Reply:
Got it. It would be good that ml codes include backup parts for the case that API communication fails.